service/api_logic/auth_strategy.py

```diff
@@ -0,0 +1,89 @@
+from abc import ABC, abstractmethod
+from typing import Dict
+import bcrypt
+import requests
+from flask import current_app, request
+from itsdangerous import URLSafeTimedSerializer
+from oauthlib.oauth2 import WebApplicationClient
+
+from database.models import User
+from dto.api_input import InputUserLoginDTO, InputUserByGoogleDTO
+from dto.api_output import OutputLogin
+from exept.exeptions import IncorrectUsernameOrEmailError, IncorrectPasswordError
+from typing import Generic, TypeVar
+
+from logger.logger import Logger
+
+T = TypeVar("T")
+
+class AuthStrategy(ABC, Generic[T]):
```

Comment:
Even though it's a strategy indeed - do not name it like that. The name must be self-explanatory and brief.

AuthHandler? AuthHelper? 

Think of some better naming for this